src/rdkafka_partition.c

```diff
@@ -2707,6 +2713,42 @@ int32_t rd_kafka_topic_partition_get_current_leader_epoch(
         return parpriv->current_leader_epoch;
 }
 
+/**
+ * @brief Sets topic id for partition \p rktpar.
+ *
+ * @param rktpar Topic partition.
+ * @param topic_id Topic id to set.
+ */
+void rd_kafka_topic_partition_set_topic_id(rd_kafka_topic_partition_t *rktpar,
+                                           rd_kafka_Uuid_t topic_id) {
+        rd_kafka_topic_partition_private_t *parpriv;
+
+        /* Avoid allocating private_t if clearing the epoch */
+        if (!rd_kafka_Uuid_cmp(topic_id, RD_KAFKA_UUID_ZERO) &&
+            !rktpar->_private)
+                return;
```

Comment:
Why this? Need more understanding on this.

Reply:
`RD_KAFKA_UUID_ZERO` is returned if `parpriv` is missing, so there's no need to create the `parpriv` to set that value. But if it's already present we need to overwrite its `topic_id` value